service/resource/cloudformation/main_stack.go

```diff
@@ -2,19 +2,27 @@ package cloudformation
 
 import (
 	"bytes"
+	"strconv"
 	"text/template"
 
-	awscloudformation "github.com/aws/aws-sdk-go/service/cloudformation"
 	"github.com/giantswarm/aws-operator/service/key"
 	"github.com/giantswarm/awstpr"
 )
 
 func newMainStack(customObject awstpr.CustomObject) (StackState, error) {
 	stackName := key.MainStackName(customObject)
+	workers := len(customObject.Spec.AWS.Workers)
+	var imageID string
+	if workers > 0 {
+		imageID = customObject.Spec.AWS.Workers[0].ImageID
+	}
+	clusterVersion := customObject.Spec.Cluster.Version
```

Comment:
FYI we have a key package to resolve information from the custom object. Lets make use of it. The code will be much nicer and eventual logic of key functions can be tested eventually.

Reply:
Done